packages/design-system/src/components/sidebar/useSidebar/index.tsx

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies.
+ */
+import React, { useCallback, useEffect, useState } from 'react';
+
+export type SidebarItem = {
+  key: string;
+  title: string;
+  children: SidebarItem[];
+  panel?: React.ReactNode;
+  icon?: React.ReactNode;
+  selectedIcon?: React.ReactNode;
+};
+
+export interface SidebarOutput {
+  activePanel: React.ReactNode;
+  selectedItemKey: string | null;
+  sidebarItems: SidebarItem[];
+  updateSelectedItemKey: (key: string | null) => void;
+  isKeyAncestor: (key: string) => boolean;
+  isKeySelected: (key: string) => boolean;
+}
+
+interface useSidebarProps {
+  data: SidebarItem[];
+}
+
+const useSidebar = ({ data }: useSidebarProps): SidebarOutput => {
+  const [selectedItemKey, setSelectedItemKey] = useState<string | null>(null);
+  const [activePanel, setActivePanel] = useState<React.ReactNode>();
+  const [sidebarItems, setSidebarItems] = useState<SidebarItem[]>([]);
+
+  useEffect(() => {
+    setSidebarItems(data);
+  }, [data]);
+
+  const matchKey = useCallback((key: string, toMatch: string) => {
```

Comment:
This function does not use any state or ref vars please consider moving it out of the component.

Reply:
This change has already been made in PR #266.